napari_animation/utils.py

```diff
@@ -31,3 +31,17 @@ def pairwise(iterable):
     a, b = itertools.tee(iterable)
     next(b, None)
     return zip(a, b)
+
+
+def check_layer_attribute_changed(value, original_value):
+    """Recursively check if a layer attribute has changed."""
+    changed = False
+    if isinstance(value, dict):
+        for key in value.keys():
+            if check_layer_attribute_changed(value[key], original_value[key]):
+                changed = True
+                break
+    else:
+        changed = not np.array_equal(value, original_value)
+
+    return changed
```

Comment:
Made this a bit more concise using early returns. I also accounted for the case where the dictionaries have different keys: it saves computations, but it also accounts for the case where `original_value` has some key that's missing from `value`.

```suggestion
def check_layer_attribute_changed(value, original_value):
    """Recursively check if a layer attribute has changed."""
    if isinstance(value, dict):
        if not isinstance(original_value, dict) or value.keys() != original_value.keys():
            return True
        for key in value.keys():
            if check_layer_attribute_changed(value[key], original_value[key]):
                return True

    return not np.array_equal(value, original_value)
```